---
title: "DAT Screen Next Generation Sequencing"
author: "Dave Speca"
date: "5/31/2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

| Sequence | Compressed | File Size | Number of reads | Source     | bam   |
| :------: | :--------: | :-------: | :-------------: | :----:     | :---: |
| Chr2R1   |            |  14 Gb    | 42,117,483      | Tail       | 33 Gb |
| Chr2R2   |            |  14 Gb    | 42,117,483      | Tail       |       |
| Chr16R1  | 14 Gb     | 41,324,213      | Tail (low) |
| Chr16R2  | 14 Gb     | 41,324,213      | Tail (low) |
| Chr18R1  | 15 Gb     | 45,940,146      | Tail       |
| Chr18R2  | 15 Gb     | 45,940,146      | Tail       |
| D2R1     | 3.3 Gb    | 15 Gb     | 44,904,397      | Liver      | 35 Gb |
| D2R2     | 3.5 Gb    | 15 Gb     | 44,904,397      | Liver      |       |

##DNA sequence analysis pipeline:
* Download using Globus
* gunzip files
* make alias:  `ln -s long_file_name.fastq shorter_name.fastq`
* count lines in file: `wc -l shorter_name.fastq`
* calculate the % of one mismatch barcodes in each file

N.B. the number of lines divided by four was equal to the reported total number of reads for all downloaded files. For Chr16, it was ~4%; all others it was ~3%

###Fastqc
* subset a smaller set of sequence data for fastqc analysis: `awk 'NR >250000 {exit} NR >=1 && NR <=250000' shorter_name.fastq > small.fastq`
* to determine the size of an individual file type: `du -h filename`

Note that the maximum attachment size for Gmail is 25 Mb. These file sizes are ~21 Mb.
Note that `du`, which stands for "disk usage" will report a different file size than `ls`.

The fastqc reports looked very similar, indicating reads of high quality. The following items were flagged:
1. "per base sequence content" at the very end of the reads. UCLA looked at the report and explained that this was because A's were trimmed in the process of removing adapter sequences. Illumina libraries use an A-tailing technique. Removal of A's near the end of the sequence is a conservative approach. No need for further processing.
2. "Kmer content" at the beginning of the sequence. Not a problem either. This is also a consequence of how Illumina libraries are made. There is a random priming step, but the annealling of the primers is not random. But UCLA said that this should not be a problem.

###Injury prevention

Illumina Reads 1 and 2 are arranged in the same order. Downstream alignment software that pairs these reads makes this assumption. Therefore, care must be taken to prune the sequence(s) of both files in the same way. To prevent screwups, Julin recommended that I write-protect the original fastq files using `chmod u-w filename.fastq`, which is what I did.

###Homebrew and bwa installation

I installed Homebrew (a Mac package manager) using this command:

`/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"`

then, to install bwa:

`brew tap homebrew/science`
`brew install bwa`
`bwa` # this brought up the reference manual, indicating proper installation.

###Download mouse reference sequence

`brew install wget`

`wget http://hgdownload.cse.ucsc.edu/goldenPath/mm10/bigZips/chromFa.tar.gz`

During the download, data transfer stopped completely for some reason. Instead of starting over, I did this:

Press `control` + `c` to stop the process and then added -c to the above command:

`wget -c http://hgdownload.cse.ucsc.edu/goldenPath/mm10/bigZips/chromFa.tar.gz`

This seemed to pick up where things left off, I hope.

This tarball came as a file of individual chromosomes. We need to unzip them, concatenate them, and then get rid of the individual chromosome files:

`tar zvfx chromFa.tar.gz`
`cat *.fa > mm10.fa`
`rm chr*.fa`

###bwa

Make index:
`bwa index mm10.fa`

Indexing took about 45 minutes to run.

`bwa mem -t 4 mm10.fa DBAR1.fastq DBAR2.fastq > DBA.sam`

Aligning took about 90 minutes to run, using -t 4 (cores), which consumed about 50% of the available CPU (although it was running at ~400%).

###samtools

`brew install samtools`

Convert .sam to .bam:
`samtools view -S -b DBA.sam > DBA.bam`

Converting to .bam took ~30 minutes

Sort .bam:
`samtools sort DBA.bam -o DBA.sort.bam`

Sorting took ~30 minutes.

Create index file:
`samtools index DBA.sort.bam`

Making index took ~10 minutes.


###Chromosome 2-specific information:
The critical region is defined by the markers D2Mit156 and D2Mit379
D2Mit156L primer: ACTGGGGAGACTAAATGGGG (mm10 chr2:57081653)
D2Mit379R primer: ACCCTAACTTGCACTCTGAAGG (mm10 chr2:68092852)

Subset DBA.sort.bam file:
(count lines first)
`samtools view DBA.sort.bam chr2:57081653-68092852 | wc -l`
457128
`samtools view Chr2.sort.bam chr2:57081653-68092852 | wc -l`
449619

### Subset sample for testing

`samtools view DBA.sort.bam chr2:57081653-68092852 > DBA.sort.chr2.test.bam`
`wc -l DBA.sort.chr2.test.bam`
457128 (same as wc -l above). Note that this did not include '-bh' (b=bam file, h=include header)
`samtools view -bh DBA.sort.bam chr2:57081653-68092852 > DBA.sort.chr2.bam`
`wc -l DBA.sort.chr2.bam`
106742

###Add Read Groups

`brew update`
`brew install cmake`
`cd /usr/local/bin`
`ls` # Hey, there's lots of stuff in there!
`git clone --recursive https://github.com/ekg/bamaddrg.git`
`cd bamaddrg`
`make`

Note that during this install, there were hundreds of warnings such as:
"/Users/Dave/bamaddrg/bamaddrg/bamtools/src/api/BamWriter.h:19:1: warning: class
      'BamAlignment' was previously declared as a struct [-Wmismatched-tags]""

Because the executable file (bamaddrg) was located in the /usr/local/bin/bamaddrg directory, I need to set an extra path:

`vi $HOME/.bash_profile`
`export PATH=$PATH:/usr/local/bin/bamaddrg`
Hit the `ESC` key. Type ":" (a colon). Type "x" and hit return to save file and close vi editor.
`source $HOME/.bash_profile`
`echo $PATH`

Result:
/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin/bamaddrg
This looks like I added the extra and correct path!

`samtools view DBA.sort.chr2.bam | head`

Result:
J00153:114:HJT2YBBXX:5:1222:14225:39858	163	chr2	57081688	60	41M22D109M	=	57081751	210	TAAGAATAACTAAATCCTGAATATGAAATACAAAATTTGGTTGTGTGTGTGTGTGTGTGTGTGTGTGTGTAATCGGTTGCATGGAAGAGTAGCAGTGTTCATTGGCTTTTTGAACACCTGATGTAAAGTTCAGTTTAAGAAGAAAACCCT	AAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJJJFJJJJJJFFJFJ<A<7<A<AJJFJJFJA77AJFJJJ<AAF<<<FFJJJJJFJJJJFAFFJJJJFAA<AA-<FFF7FFFFFFJ<AFAJF	NM:i:22	MD:Z:41^TGTGTGTGTGTGTGTGTGTGTG109	AS:i:122	XS:i:37

Now add Read Groups:

`bamaddrg -b DBA.sort.chr2.bam -s DBA -r DBA > DBA.sort.chr2.RG.bam` 
`samtools view DBA.sort.chr2.RG.bam | head`

Result:
J00153:114:HJT2YBBXX:5:1222:14225:39858	163	chr2	57081688	60	41M22D109M	=	57081751	210	TAAGAATAACTAAATCCTGAATATGAAATACAAAATTTGGTTGTGTGTGTGTGTGTGTGTGTGTGTGTGTAATCGGTTGCATGGAAGAGTAGCAGTGTTCATTGGCTTTTTGAACACCTGATGTAAAGTTCAGTTTAAGAAGAAAACCCT	AAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJJJFJJJJJJFFJFJ<A<7<A<AJJFJJFJA77AJFJJJ<AAF<<<FFJJJJJFJJJJFAFFJJJJFAA<AA-<FFF7FFFFFFJ<AFAJF	NM:i:22	MD:Z:41^TGTGTGTGTGTGTGTGTGTGTG109	AS:i:122	XS:i:37	RG:Z:DBA

###Concatenate and index the two subsetted files:
`bamaddrg -b DBA.sort.chr2.bam -s DBA -r DBA \ bamaddrg -b Chr2.sort.chr2.bam -s Chr2 -r Chr2 > Chr2.test.bam`
`samtools index Chr2.test.bam`

###Use freebayes to make .vcf file
`freebayes -f mm10.fa Chr2.test.bam > Chr2.test.vcf`
Note that there was no index for mm10, so freebayes made it: mm10.fai

###Use R to look at the vcf file!


###Questions:
1. File management and how to prevent creation of files during each step of the process.
2. Related to above, how to create a pipeline. RAM issues for running a pipeline.
3. I was confused about the format for the Read Group.
4. How to access statistics about coverage, number of unaligned reads, etc.?
